static/app/components/teamSelector.tsx

```diff
@@ -303,7 +316,13 @@ function TeamSelector(props: Props) {
   );
 
   function getOptions() {
-    const filteredTeams = teamFilter ? teams.filter(teamFilter) : teams;
+    const memberSettingsFilteredTeams = memberTeamsOnly
+      ? teams.filter((team: Team) => team.isMember)
+      : teams;
+    const filteredTeams = teamFilter
+      ? memberSettingsFilteredTeams.filter(teamFilter)
+      : memberSettingsFilteredTeams;
```

Comment:
I know you're expanding on what was already here, but ternaries become difficult to read outside of simple use cases. Consider using `if` instead.

Additionally, this performs the filter on `teams` every time `getOptions` is called. Can these values can be derived outside of the `getOptions()` closure?